src/staking-router/staking-router.service.ts

```diff
@@ -18,39 +19,88 @@ export class StakingRouterService {
     protected readonly keysApiService: KeysApiService,
   ) {}
 
-  /**
-   * Return staking module data and block information
-   */
-  public async getStakingModulesData(): Promise<{
-    stakingModulesData: StakingModuleData[];
-    blockHash: string;
-    blockNumber: number;
-  }> {
+  protected stakingRouterCache: Record<number, StakingModuleData> = {};
+
+  async getOperatorsAndModules() {
     const { data: operatorsByModules, meta: operatorsMeta } =
       await this.keysApiService.getOperatorListWithModule();
 
+    return { data: operatorsByModules, meta: operatorsMeta };
+  }
+
+  async fetchUnusedKeys(moduleAddresses: string[]) {
     const { data: unusedKeys, meta: unusedKeysMeta } =
-      await this.keysApiService.getUnusedKeys();
+      await this.keysApiService.getUnusedKeys(moduleAddresses);
 
-    const blockHash = operatorsMeta.elBlockSnapshot.blockHash;
-    const blockNumber = operatorsMeta.elBlockSnapshot.blockNumber;
+    return { data: unusedKeys, meta: unusedKeysMeta };
+  }
+
+  /**
+   * Return staking module data and block information
+   */
+  async getStakingModulesData(
```

Comment:
This is essentially a caching layer around the method of retrieving unused keys. perhaps we should reflect that in the name?